src/layout/layout.rs

```diff
@@ -889,12 +1149,138 @@ impl Element {
     }
 }
 
+#[derive(Debug)]
+struct StrengthSet {
+    pub spacer_size_equality: f64,
+    pub proportional_scaling_equality: f64,
+    pub fixed_value_equality: f64,
+    pub min_max_inequality: f64,
+    pub length_equality: f64,
+    pub percentage_equality: f64,
+    pub ratio_equality: f64,
+    pub min_max_equality: f64,
+    pub proportional_grower: f64,
+    pub grower: f64,
+    pub space_grower: f64,
+}
```

Comment:
Given that these are basically constant and not user configurable (YAGNI for now?):
```rust
mod strengths {
    /// The strength used to constrain spacers to approximately equal size
    /// 
    /// ```plain
    /// ┌      ┐┌───┐┌      ┐┌───┐┌       ┐
    ///    eq   │   │   eq   │   │   eq
    /// └      ┘└───┘└      ┘└───┘└       ┘
    /// ```
    SPACER_SIZE_EQ: f64 = REQUIRED - 1.0;
    PROPORTIONAL_SCALING_EQ: f64 = REQUIRED - 1.0;
    FIXED_SIZE_EQ: f64 = REQUIRED / 10.0;
    MIN_SIZE_GE: f64 = STRONG * 10.0;
    MAX_SIZE_LE: f64 = STRONG * 10.0;
    LENGTH_SIZE_EQ: f64 = STRONG / 10.0;
    PERCENTAGE_SIZE_EQ: f64 = MEDIUM * 10.0;
    RATIO_SIZ_EQ: f64 = MEDIUM;
    MIN_SIZE_EQ: f64 = MEDIUM / 10.0;
    MAX_SIZE_EQ: f64 = MEDIUM / 10.0;
    PROPORTIONAL_GROWER: f64 = WEAK * 10.0
    GROWER: f64 = WEAK * 10.0;
    SPACE_GROWER: f64 = WEAK / 10.0;
}

Reply:
Good idea! I was hoping to make it configurable for easier comparison between `main` and this branch but that never came to fruition.